src/components/Input.js

```diff
@@ -49,11 +68,18 @@ class Input extends React.PureComponent {
             onFocus={onFocus}
             onKeyPress={onKeyPress}
             onChange={onChange}
+            ref={(input) => { this.input = input; }}
           />
-          {error && <ErrorIcon className="drui-inputError__errorIcon" />}
+          {initialType === 'password' && !error && (
```

Comment:
Ok, 2 things here.
First of all, lets make this component accept a prop called e.g. `passwordVisibilityToggle` with a boolean value that will indicate if we should show that password visibility toggle button or not (ofc only when `this.props.type === 'password'`. It will also make `this.initialType` useless.
Also, we can't hide that button when an error occurs, because this will be probably the moment when the users are going to need it the most ;)